schema/src/bindings/typescript.ts

```diff
@@ -40,13 +40,9 @@ export const generateTypeDefinitions = async (): Promise<string> => {
 
 ${autogeneratedHeader('build:ts', path.basename(__filename), '//')}
 
-type Null = null
-type Boolean = boolean
 type Integer = number
-type Number = number
-type String = string
-type Object = Record<string, unknown>
-type Primitives = undefined | null | boolean | number | string | Array<unknown> | Object
+type ObjectType = Record<string | number, unknown>
```

Comment:
Can we make `type ObjectType = Record<string, unknown>`? This is consistent with the definition in Rust:

https://github.com/stencila/stencila/blob/7d3016a413859256ed4c553adc08bdca9a87807f/rust/schema/src/prelude.rs#L66-L69

This of leaving it as it is, is that an object with a number key will be valid in TypeScript but then will fail to deserialize to memory in Rust.

To be consistent with the other primitives, I think I'd prefer to just use `Record<string, unknown>` everywhere instead of introducing `ObjectType`.

Also, to be consistent with Rust, and with the union type for `Node` we should de-pluralize to `Primitive`

Reply:
I've made the changes.
Couple questions though:
- Should Array inside `Primitive` be `Array<Primitive>` instead of `Array<unknown>`?
- Can we restrict the Object type inside the `Primitive` type to something more meaningful, rather than being compatible with any and all schema nodes?

Reply:
Yes ideally, they would be as in Rust: `Array<Primitive>` and `Record<String, Primitive>` but I wasn't sure  / wasn't able (?) to define recursive types in TS. Please do so if possible.

Reply:
Done. There's a weird quirk with recursive types in that:

```diff
type Primitive =
  | undefined
  | null
  | boolean
  | number
  | string
  | Array<Primitive>
- | Record<string, Primitive> // Doesn't work
+ | { [key: string]: Primitive } // Works
```

¯\\\_(ツ)\_/¯

Reply:
One last suggestion while we're at it...I don't think `undefined` needs to be in `Primitive`